modules/hpa/main.tf

```diff
@@ -0,0 +1,757 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+resource "kubernetes_manifest" "clusterrole_scaling_prometheus" {
+  manifest = {
+    "apiVersion" = "rbac.authorization.k8s.io/v1"
+    "kind" = "ClusterRole"
+    "metadata" = {
+      "name" = "scaling-prometheus"
+    }
+    "rules" = [
+      {
+        "apiGroups" = [
+          "",
+        ]
+        "resources" = [
+          "nodes",
+          "nodes/proxy",
+          "services",
+          "endpoints",
+          "pods",
+        ]
+        "verbs" = [
+          "get",
+          "list",
+          "watch",
+        ]
+      },
+      {
+        "apiGroups" = [
+          "",
+        ]
+        "resources" = [
+          "configmaps",
+        ]
+        "verbs" = [
+          "get",
+        ]
+      },
+      {
+        "nonResourceURLs" = [
+          "/metrics",
+        ]
+        "verbs" = [
+          "get",
+        ]
+      },
+      {
+        "apiGroups" = [
+          "",
+        ]
+        "resourceNames" = [
+          "scaling-prometheus-headless:9090",
+        ]
+        "resources" = [
+          "services/proxy",
+        ]
+        "verbs" = [
+          "*",
+        ]
+      },
+      {
+        "nonResourceURLs" = [
+          "/api/v1/namespaces/sn-system/services/scaling-prometheus-headless:9090/proxy/*",
+        ]
+        "verbs" = [
+          "*",
+        ]
+      },
+    ]
+  }
+}
+
+resource "kubernetes_manifest" "serviceaccount_sn_system_scaling_prometheus" {
+  manifest = {
+    "apiVersion" = "v1"
+    "kind" = "ServiceAccount"
+    "metadata" = {
+      "annotations" = {
+        "iam.gke.io/gcp-service-account" = "pulsar-prometheus@sncloud-dev-zxc.iam.gserviceaccount.com"
```

Comment:
The value for this annotation should be a variable that you pass to the module

Reply:
seems added when I tested in my dev env while I've try deploy this in test env GCP and AWS cluster and works fine, so this should be needed.